### ISSUE TITLE
Build problem - DatabaseWorkerPool

### ISSUE THREAD

**Issue by Toooorch** [TrinityCore/TrinityCore#19967]
**Description:**
Build failed.

I have fresh os and latest mariadb installed through homebrew. (mariadb 10.2.6)

**Current behaviour:**
I can not build the core.

**Expected behaviour:**
Successfully compiled core

**Steps to reproduce the problem:**

1. clone latest commit 346e498707fe5226f61bc3489cb255f891cc777f
2. configure cmake - xcode build (https://gist.github.com/Toooorch/93140e27d999698446c875cc0fe35251)
3. build as release/debug

>** BUILD FAILED **

> The following build commands failed:
> 	CompileC /Users/torch/desktop/TrinityServers/TrinityCore/build/src/server/database/TrinityCore.build/Debug/database.build/Objects-normal/x86_64/DatabaseWorkerPool.o src/server/database/Database/DatabaseWorkerPool.cpp normal x86_64 c++ com.apple.compilers.llvm.clang.1_0.compiler

when I try to build it in x-code. It gives an error in **DatabaseWorkerPool.cpp**
<img width="1139" alt="screen shot 2017-06-29 at 00 15 35" src="https://user-images.githubusercontent.com/4589728/27662905-26418586-5c60-11e7-9d0e-4b0df2fada45.png">

cmake output https://gist.github.com/Toooorch/2b50bc46fb540ae02109f4e4739b81b3

**Branch(es):**
master 

**TC rev. hash/commit:** 
346e498707fe5226f61bc3489cb255f891cc777f

**Operating system:** 
Mac OS X

**Comment by Shauren** [comment]
> Successfully compiled core

That is not the expected result since you are not using MySQL

**Comment by Toooorch** [comment]
oh ok. Sorry for disrupting but I throught that MariaDB is supported because I saw some dumbs from it on repo/sql/. I will try it with MySQL tomorrow...

**Comment by asd567800** [comment]
using mysql 5.7.xxx client or server sdk will work. And server is mariadb 10.2

**Comment by Shauren** [comment]
This shouldn't be too surprising - it doesn't even work with mysql connector (only libmysqlclient is supported)

**Comment by Aokromes** [comment]
debian 9 works fine with apt-get mariadb.

**Comment by Shauren** [comment]
mariadb server sure but its libmysqlclient-dev that is used to build TC

**Comment by Aokromes** [comment]
vm compiled with:

apt-get install git cmake make gcc g++ libmariadbclient-dev libssl1.0-dev libbz2-dev libreadline-dev libncurses-dev libboost-all-dev mysql-server p7zip

**Comment by Toooorch** [comment]
I just figured out where the problem is.

After OS reinstall I just installed dependencies, as usual, using homebrew BUT MariaDB now not including the libmysqlclient or those dev libraries. 
It is funny because this dylibs was included in older versions and now there is no way to get those dev libraries on Mac except downloading the older version of MariaDB (10.1 is just fine). It took one five minutes to build that on ubuntu...

And as Shauren said - it doesn't even work with connectors...

Thank you, guys. I am sorry for my local problem. Hope I will be able to contribute because I have one charr on an official server...
